8.0 distribution list imp

JonathanNEMRY opened the pull request:
Improve feature 'more filter' by adding this option into the 'alternative_object' mode
Fix a bug related to this alternative_object
Improve test to cover those changes

Commit:
[CHG] give the possibility to add an alternative 'more_filter' option just as 'more_filter' do with 'main_object'
[FIX] use the right field to compute alternative ids
[CHG] improve test to cover the previous bug and new feature too

Commit:
[distribution_list][CHG] review code

JonathanNEMRY commented:
Any news about this PR?

lmignon commented on distribution_list/distribution_list.py:
```diff
@@ -292,8 +293,12 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
+                    domain_alternative_objects = \
+                        [('id', 'in', res_ids),
+                         (alternative_object, '!=', False)]
+                    if context.get('alternative_more_filter', False):
```
``` python
if context.get('alternative_more_filter')
```

lmignon commented on distribution_list/distribution_list.py:
```diff
@@ -272,11 +274,10 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
-                domains = [('id', 'in', res_ids)]
+                domain_main_objects = [('id', 'in', res_ids),
+                                       (main_object, '!=', False)]
                 if context.get('more_filter', False):
```
``` python
if context.get('more_filter')
```

Commit:
[distribution_list][FIX] case where distribution list has no filter then bridge field should be equals to 'id'
because there is no more need to convert

Commit:
[distribution_list]
[FIX] id instead of bridge_field: wrong place
[CHG] notebook/search view and add a note into distribution list

Olivier-LAURENT committed:
Update distribution_list_view.xml

lmignon commented on distribution_list/distribution_list.py:
```diff
@@ -292,8 +296,12 @@ def get_complex_distribution_list_ids(self, cr, uid, ids, context=None):
                         else:
                             result_ids .append(val[main_object])
                 if alternative_object:
```
Is-it possible to split the computation of main_object and alternative_object into 2 sub methods to improve the readability and the modularity?

Olivier-LAURENT committed:
[CHG] code review: more modularity

JonathanNEMRY commented:
:+1:

lmignon commented on distribution_list/distribution_list.py:
```diff
@@ -235,86 +238,69 @@ def get_ids_from_distribution_list(self, cr, uid, ids, safe_mode=True,
         return list(set_included_ids)
 
+    def _get_ids(self, cr, uid, ids, model, fld, dom, sort, context=None):
+        res_ids = []
```
fld means field_name?, dom domain?

Cedric-Pigeon commented:
:+1:

lmignon commented:
Nice refactoring!  A description would be nice for the _get_ids method. :+1:

Olivier-LAURENT committed:
[CHG] add a docstring
